google/cloud/storage/internal/object_write_streambuf.cc

```diff
@@ -200,8 +199,10 @@ void ObjectWriteStreambuf::FlushRoundChunk(ConstBufferSequence buffers) {
     // even if no "final chunk" is sent.  The resumable upload classes know how
     // to deal with this mess, so let's not duplicate that code here.
     auto actual_next_byte = upload_session_->next_expected_byte();
-    if (actual_next_byte < expected_next_byte &&
-        actual_next_byte < first_buffered_byte) {
+    if (actual_next_byte < expected_next_byte) {
+      // TODO(#8559) - if actual_next_byte < first_buffered_byte and there is
+      //   enough space in the buffer we could save the bytes for a future
+      //   write.
       std::ostringstream error_message;
       error_message << "Could not continue upload stream. GCS requested byte "
```

Comment:
Does this error message need to be updated?
If actual next byte is less than expected next byte, that means some bytes have not been committed and therefore might not be uploaded yet?